chia/util/augmented_chain.py

```diff
@@ -47,8 +47,7 @@ def add_extra_block(self, block: FullBlock, block_record: BlockRecord) -> None:
 
     def remove_extra_block(self, hh: bytes32) -> None:
         if hh in self._extra_blocks:
-            block_record = self._extra_blocks.pop(hh)[1]
-            del self._height_to_hash[block_record.height]
```

Comment:
it's not obvious why you removed this line. Don't we need to clear the `_height_to_hash` entry?

Reply:
no, the point of this pr is for the augmented chain to contain the alternate height_to_hash, remove extra block is called once we validated the block but no necessarily when we added it to the height to hash

Reply:
Right, in some cases we sync against a new reorged peak, and we need to distinguish between the fork's height-to-hash and the current main chain's. However, sometimes we sync for a very lon time and never really change fork, we just sync against the newest and heaviest block. In this case (I believe) we keep updating our own peak, and blocks are actually added to the underlying main cnain as we validate them.

In the second scenario, we would probably want to avoid storing the fulll height-to-hash in memory. It can get quite big.

Reply:
np, we can remove from the augmented if it is contained in the underlying, which now we can do by checking contains, that would only be true if the latest block from the new fork is already higher then the previous chain we where on and is the current peak

Reply:
the scenario I have in mind is a long sync, say, from scratch. And technically, it's not the height-to-hash itself that uses so much space (not yet) but if we would need to keep all the blocks in the agumented blockchain as well (because the height-to-hash lookup suggest they should be there), that would use too much space.

Reply:
(feel free to resolve this conversation when you've seen it)

Reply:
the blocks themselves get removed as they get validated since they are added to the cache of the underlying, the hight_to_hash of the underlying only gets updated with said blocks when we move past the peak of the underlying chain and we switch to the fork in the underlying